test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,488 @@
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count: {}, write count: {} after SSS failover after sleep", currentReadCount1, currentWriteCount1);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount1 > currentWriteCount2);
+        //ensure reads are happening
+        assertTrue(currentReadCount1 > currentReadCount2);
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+        log.info("Scaling down controller instances from 3 to 2");
+
+        currentWriteCount2 = eventWriteCount.get();
+        currentReadCount2 = eventReadCount.get();
+
+        log.info("Read count: {}, write count: {} after controller failover after sleep", currentReadCount2, currentWriteCount2);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS, controller to 1 instance each.
+        segmentStoreInstance.scaleService(1, true);
+        controllerInstance.scaleService(1, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(new Random().nextInt(50000) + ZK_DEFAULT_SESSION_TIMEOUT);
+        log.info("Scaling down  to 1 controller, 1 SSS instance");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Stop write flag status: {}, stop read flag status: {} ", stopWriteFlag, stopReadFlag);
+        log.info("Event data: {}, read count: {}", eventData.get(), eventReadCount.get());
+        log.info("Read count: {}, write count: {} with SSS and controller failover after sleep", currentReadCount1, currentWriteCount1);
+    }
+
+
+
+    private CompletableFuture<Void> startWriting(final AtomicLong data, final EventStreamWriter<Long> writer,
+                                                 final AtomicBoolean stopWriteFlag, final AtomicLong eventWriteCount) {
+        return CompletableFuture.runAsync(() -> {
+            while (!stopWriteFlag.get()) {
+                try {
+                    long value = data.incrementAndGet();
+                    Thread.sleep(100);
+                    log.debug("Event write count before write call {}", value);
+                    writer.writeEvent(String.valueOf(value), value);
+                    log.debug("Event write count before flush {}", value);
+                    writer.flush();
+                    eventWriteCount.getAndIncrement();
+                    log.debug("Writing event {}", value);
+                } catch (Throwable e) {
+                    log.error("Test exception writing events: ", e);
```

Comment:
You should find a way to make this fail the test. Perhaps setting a an exception on member variable that is checked by the main thread.